### ISSUE TITLE
Make widget SMALLER

### ISSUE THREAD

**Issue by Comfunc** [mvysny/photocloud-frame-slideshow#192]
I know the widget can be made bigger, but I'd like the option to make it _**smaller**_. 

_I want to make it 1/3rd size,_ but 1/2 size may also be useful.

The reason is that on a big screen, the widget is over twice as big as it needs to be to show its text, so it unnecessarily overlays and spoils the photo image.

**Comment by Comfunc** [comment]
Sorry to be slow, but I've looked everywhere and can't see where or how I post a bounty on this?

**Comment by rajdude** [comment]
Yeah, the last time I asked Martin to fix something was like 4 or 5 years ago and I used bountysource dot com. That website is gone now. I also may want to post a bounty on an issue I just opened. 
BTW, since it has been 2 weeks and Martin has not replied to you, I am a little worried about this app. This is the ONLY app which has the features I want and use. I hope it stays alive.

**Comment by rajdude** [comment]
> I know the widget can be made bigger, but I'd like the option to make it _**smaller**_.
> 
> _I want to make it 1/3rd size,_ but 1/2 size may also be useful.
> 
> The reason is that on a big screen, the widget is over twice as big as it needs to be to show its text, so it unnecessarily overlays and spoils the photo image.

By the way, I am not sure why you feel that the HUD widget is too big. I have a 85" TV which I use this app on (running on nVidia Shield). I disabled the clock (no need). Then I actually made the widget bigger (2x) since it was not showing the whole path of my photo, along with its name.

You are talking about the text on the lower left corner of the screen, correct?

I do not see any other widget while the slide-show is running. Just the photo is there on the whole TV screen (and the file name on the lower left, which I do want).

**Comment by Comfunc** [comment]
I'm glad I'm not the only one thinking along these lines.

Basically, the current model of Android one-off payment software
development, is broken. We need another model which is similarly based on
the idea of massive numbers allowing a low payment, but the payment needs
to be repeating yearly in order to allow support.

Or another model is that each major version is available for a one-off
cost, but after, say 3 Android OS versions, it's behaviour is patchy or
broken, and you need to pay for a new version. After all, if you're still
using that app after 3 OS upgrades, it must be useful to you.

On Windows I use Directory Opus and Applian Capture Suite which run along
these lines, and it seems quite reasonable.
[sent from mobile]

**Comment by Comfunc** [comment]
Yes, I'm talking about the text in the lower left corner of the screen.

The selection of information to display is complicated. Too much, and it's
lost. Too little, and you are lost. That's why you removed the clock - it's
not necessary (ie strictly necessary for YOUR purposes) so is a distraction.

The aspect ration of my photos is less wide than the aspect ratio of my
screen. So I have a couple of totally reasonable black bars down the side
of even landscape photos.

I want the widget to fit in this black bar. There's plenty of space and I
don't want it plastered over the bottom of my photo. Those images represent
many hours of careful work. They are not enhanced by having text written
across them. And I only want to refer to that text occasionally, so it can
be small and discrete.

Actually, I worked in  software development  for many decades. My approach
was to give the user whatever trivial customisation options he wanted. The
world out there is way bigger than you can ever imagine, so it's best to
just give it the freedom to adapt in the best way it sees fit. It will
eventually boil down to some kind of stable stasis, and probably in a more
functional form than you could have imagined.
[sent from mobile]

**Comment by rajdude** [comment]
Hmm.....I see your viewpoint. I do not have a solution, but I do have some suggestions...

1. My photos are also not 16:9. They are all 4:3 aspect ratio so they are displayed on my TV as pillar boxed. But I just found a setting in this app, which zooms into the photo and then scrolls the photo up and down slowly. That fixes the pillar box photos problem (for me at least). The photo then fills the entire screen's width. But I hear you, you want the HUD (as Martin calls it) to be on the side, in those black bars and not intruding into your photos. If I were you, I would simply disable HUD. You CAN show that data if needed, by hitting the up arrow on the remote. I think on a touchscreen, a tap does it.
2. I also work in IT, and I have noticed that --- what is very obvious to one person could be totally oblivious to others! LoL. That is why so many things which seem to be no brainer to a consumer are not implemented by a software developer. I know this is not a suggestion, but unfortunately, I deal with this every day :-) Maybe a software enhancement request is in order - make the HUD vertical on the left/right side! ;-) ?

**Comment by Comfunc** [comment]
Thanks for the suggestions!

1. Perhaps I don't belong in this modern world anymore.
* I've no problem with black bars at the sides of photos or movies. I just
want it in it's orignal aspect ratio. After spending thousands of hours
taking and curating high quality photos and images, I just want them
displayed as is. Would you do the automatic cropping and  zooming around
thing with a Rembrandt?  Actually, it's a great idea. If they put portholes
in front of Guernica, then many more people could see it (altho they'd only
see a bit of it at a time as the porthole moved forward).

* I also don't have a problem with margins on tablets. When I pick up my
new tablets, I often end up skipping video because my fingers project onto
the screen. I put a case on but it still happens. Maybe the modern humans
don't need to hold onto things somehow.

2. This is so true. I managed to preserve my 'normal person' mind even
while becoming a software developer, so I could see why the users got so
upset.
As for the HUD, if I did it, I'd just make 4 fields corresponding to the 4
corners, which could be configured with any of about 4 bits of information,
and which have a few formatting and sizing options. It's really only a
single piece of code, with a placement option.  There's no point in
deciding anything specific, just let the user do what he wants,  and at the
end of the day it's only fractionally more work. The objection is "But what
if the users puts the time in BOTH top corners?" and the answer is "I don't
care".

:-)
